core/lib/zksync_core/src/lib.rs

```diff
@@ -630,6 +630,25 @@ pub async fn initialize_components(
                     Arc::new(ValidiumModeL1BatchCommitDataGenerator {})
                 }
             };
+
+        let args = CallFunctionArgs::new("getPubdataPriceMode", ()).for_contract(
+            contracts_config.executor_facet_addr,
+            zksync_contracts::executor_contract(),
+        );
+        let current_commitment_mode_eth_response = eth_client.call_contract_function(args).await?;
+
+        let current_commitment_mode =
+            L1BatchCommitDataGeneratorMode::from_tokens(current_commitment_mode_eth_response)?;
+
+        // contracts mode == server mode
+        assert_eq!(
+            current_commitment_mode,
+            state_keeper_config.l1_batch_commit_data_generator_mode,
+            "The selected L1BatchCommitDataGeneratorMode ({:?}) does not match the existing commitment mode ({:?})",
```

Comment:
Maybe replace "existing commitmend mode" with "commitment mode used on L1 contract" to make it more clear?